areas/appservice/tests/test-resources.bicep

```diff
@@ -0,0 +1,377 @@
+param sqlAdminLogin string = 'mcptestadmin'
+
+@description('SQL Server administrator password.')
+@secure()
+param sqlAdminPassword string = newGuid()
+
+@description('MySQL administrator login name.')
+param mysqlAdminLogin string = 'mcptestadmin'
+
+@description('MySQL administrator password.')
+@secure()
+param mysqlAdminPassword string = newGuid()
+
+@description('PostgreSQL administrator login name.')
+param postgresAdminLogin string = 'mcptestadmin'
+
+@description('PostgreSQL administrator password.')
+@secure()
+param postgresAdminPassword string = newGuid()
+
+@description('The SKU for the App Service Plan.')
+param appServicePlanSku string = 'B1'
+
+// Variables
+var webAppName = '${baseName}-webapp'
+var appServicePlanName = '${baseName}-plan'
+var sqlServerName = '${baseName}-sql'
+var sqlDatabaseName = '${baseName}db'
+var mysqlServerName = '${baseName}-mysql'
+var mysqlDatabaseName = '${baseName}mysqldb'
+var postgresServerName = '${baseName}-postgres'
+var postgresDatabaseName = '${baseName}postgresdb'
+var cosmosAccountName = '${baseName}-cosmos'
+var cosmosDatabaseName = '${baseName}cosmosdb'
+
+// App Service Plan
+resource appServicePlan 'Microsoft.Web/serverfarms@2023-01-01' = {
+  name: appServicePlanName
+  location: location
+  sku: {
+    name: appServicePlanSku
+    tier: 'Basic'
+    size: appServicePlanSku
+    family: 'B'
+    capacity: 1
+  }
+  properties: {
+    perSiteScaling: false
+    elasticScaleEnabled: false
+    maximumElasticWorkerCount: 1
+    isSpot: false
+    reserved: false
+    isXenon: false
+    hyperV: false
+    targetWorkerCount: 0
+    targetWorkerSizeId: 0
+    zoneRedundant: false
+  }
+}
+
+// Web App
+resource webApp 'Microsoft.Web/sites@2023-01-01' = {
+  name: webAppName
+  location: location
+  kind: 'app'
+  properties: {
+    enabled: true
+    hostNameSslStates: [
+      {
+        name: '${webAppName}.azurewebsites.net'
+        sslState: 'Disabled'
+        hostType: 'Standard'
+      }
+      {
+        name: '${webAppName}.scm.azurewebsites.net'
+        sslState: 'Disabled'
+        hostType: 'Repository'
+      }
+    ]
+    serverFarmId: appServicePlan.id
+    reserved: false
+    isXenon: false
+    hyperV: false
+    vnetRouteAllEnabled: false
+    vnetImagePullEnabled: false
+    vnetContentShareEnabled: false
+    siteConfig: {
+      numberOfWorkers: 1
+      acrUseManagedIdentityCreds: false
+      alwaysOn: false
+      http20Enabled: false
+      functionAppScaleLimit: 0
+      minimumElasticInstanceCount: 0
+    }
+    scmSiteAlsoStopped: false
+    clientAffinityEnabled: true
+    clientCertEnabled: false
+    clientCertMode: 'Required'
+    hostNamesDisabled: false
+    containerSize: 0
+    dailyMemoryTimeQuota: 0
+    httpsOnly: false
+    redundancyMode: 'None'
+    storageAccountRequired: false
+    keyVaultReferenceIdentity: 'SystemAssigned'
+  }
+}
+
+// SQL Server
+resource sqlServer 'Microsoft.Sql/servers@2023-05-01-preview' = {
```

Comment:
We should consider if we use the sql from the /sql folder.  I'm not sure if the deploy-testresources is setup to do dependencies, but I would like to not have to provision to sql servers and databases unless absolutely necessary.